src/PanoImageRenderer/WebGLUtils.ts

```diff
@@ -0,0 +1,207 @@
+import agent from "@egjs/agent";
+import { TypedArray } from "../types";
+
+const WEBGL_ERROR_CODE = {
+  "0": "NO_ERROR",
+  "1280": "INVALID_ENUM",
+  "1281": "INVALID_VALUE",
+  "1282": "INVALID_OPERATION",
+  "1285": "OUT_OF_MEMORY",
+  "1286": "INVALID_FRAMEBUFFER_OPERATION",
+  "37442": "CONTEXT_LOST_WEBGL"
+};
+
+let webglAvailability: boolean | null = null;
+let MAX_TEXTURE_SIZE_FOR_TEST: number | null = null;
+
+export default class WebGLUtils {
+  public static createShader(gl: WebGLRenderingContext, type: number, source: string) {
+    const shader = gl.createShader(type)!;
+
+    gl.shaderSource(shader, source);
+    gl.compileShader(shader);
+    const success = gl.getShaderParameter(shader, gl.COMPILE_STATUS);
+
+    if (success) {
+      return shader;
+    } else {
+      // eslint-disable-next-line
+      console.error(gl.getShaderInfoLog(shader));
+    }
+    return null;
+  }
+
+  public static createProgram(gl: WebGLRenderingContext, vertexShader: WebGLShader, fragmentShader: WebGLShader) {
+    const program = gl.createProgram()!;
+
+    gl.attachShader(program, vertexShader);
+    gl.attachShader(program, fragmentShader);
+    gl.linkProgram(program);
+
+    gl.detachShader(program, vertexShader);
+    gl.detachShader(program, fragmentShader);
+    gl.deleteShader(vertexShader);
+    gl.deleteShader(fragmentShader);
+
+    const success = gl.getProgramParameter(program, gl.LINK_STATUS);
+
+    if (success) {
+      return program;
+    }
+
+    gl.deleteProgram(program);
+    return null;
+  }
+
+  public static initBuffer(gl: WebGLRenderingContext, target: number /* bind point */, data: TypedArray, itemSize: number, attr?: number) {
+    const buffer = gl.createBuffer()!;
+
+    gl.bindBuffer(target, buffer);
+    gl.bufferData(target, data, gl.STATIC_DRAW);
+
+    if (buffer) {
+      (buffer as any).itemSize = itemSize;
```

Comment:
I've used any not because `itemSize` is not enlisted on the `WebGLBuffer`, but as it's a custom property that we use.
It is better to use those properties not like that, but I didn't want to refactor a whole source code on this work.
I think it should be divided into a different issue.